rollup.config.js

```diff
@@ -24,49 +24,73 @@ const LEGAL_TEXT = `/*
 * Attribution and license information can be found at https://github.com/MyPureCloud/client-app-sdk/blob/master/README.md
 */`;
 
-const baseRollupConfig = {
+// Packages to exclude from esm/cjs bundles
+const EXTERNAL_DEPS = ['@babel/runtime', 'query-string'];
+
+const buildRollupConfig = (config) => Object.assign({
     input: './src/index.js',
     cache: false,
     plugins: [
         commonjs(),
         resolve(),
-        json(),
-        babel({
-            exclude: 'node_modules/**'
-        })
+        json()
     ]
-};
+}, config);
 
-const buildRollupConfig = (config) => {
-    return Object.assign({}, baseRollupConfig, config);
-};
+const umdBabelOutputPlugin = getBabelOutputPlugin({
+    // Need to specify this for a umd output format
+    // https://github.com/rollup/plugins/tree/master/packages/babel#using-formats-other-than-es-modules-or-commonjs
+    allowAllFormats: true,
+    presets: ['@babel/env']
```

Comment:
Is `modules: false` not needed here?  I assumed we always wanted rollup to deal with imports, but maybe it has already happened by the time babel runs in the umd case?

Reply:
Oh man, do I have an answer for you on this one. So, I had to dig a bit for this one to be 100% positive (thank you garbage babel docs). The default behavior is `modules: "auto"`. Modules auto defaults to `false` if the current process is known to support ESM syntax. Relevant docs you'll only find in this PR: https://github.com/babel/babel/pull/8485. So there is actually a flag `supportsStaticESM` that can be specified when running babel directly. It just so happens that `@rollup/plugin-babel` supplies `supportsStaticESM: true` to babel in their implementation, [(see here)](https://github.com/rollup/plugins/blob/440dbc4f53c8ded33f2ad58f10ea647b30807435/packages/babel/src/index.js#L45) so we can safely drop `modules: false` from all of our builds. Thank you for coming to my TED talk.

I'm not opposed to just specifying `modules: false` everywhere to avoid confusion, what do you think?

Reply:
https://media.giphy.com/media/3ELtfmA4Apkju/giphy.gif

Reply:
haha .. thank you for digging that up.  I would be good with either always specifying and/or leaving a comment.  I feel like that might bite us in the future at some point.

Reply:
Yeah, I'm good with just specifying it everywhere as I don't want to have to go down that rabbit hole again 😆